app/forms/thredded/topic_form.rb

```diff
@@ -36,6 +36,7 @@ def save
       ActiveRecord::Base.transaction do
         topic.save!
         post.save!
+        UserTopicReadState.read_on_first_post!(user, topic) if topic.previous_changes.include?(:id)
```

Comment:
I see this is not strictly necessary for private topics because the topic is displayed after creating it, but I think it'd better to not have the code rely on this.